docs/architecture.md

```diff
@@ -111,6 +111,8 @@ class PulpMyPluginRepositoryContext(PulpRepositoryContext):
         return body
 ```
 
+Note that specifier `>=x.y.z` doesn't include `x.y.z.dev`, which  means that when adding or removing a feature from a plugin, that change must first be released before it's constrains can be realiably tested in the CLI.
+
```

Comment:
Great addition. Let's spice it up a little:
```suggestion
!!! note
    The specifier `>=x.y.z` doesn't include `x.y.z.dev` according to PEP 440.
    Therefore, when adapting to an unreleased feature change from a plugin, you need to specify the prerelease part of the version explicitly.
    However `>=x.y.z.dev` is never unambiguous in the current Pulp versioning practice.
    Once that change is released please reset the constraint to the plain `x.y.z` schema.

```

Note that the meaning is somewhat different, stating that you _can_ test unreleased features in the CLI, but the "dev" version part is never unambiguous.

Maybe it's time to look into `setuptools-scm` for Pulp. We'd get automatic `dev<n>` versions.

Reply:
Thanks, that clarifies what/why to do.
The idea of using `setuptools-scm` is to have unique dev identifiers for each commit, so pre-release specifiers can't live unnoticed?

Reply:
I believe the idea is to derive the current version of the project from git. But i just realized this will not work, since we do not tag our releases on the main branch. So last "tag + n commits" is meaningless. But it would have been nice to make assertions like `specifier>="1.2.3.dev15"`.